bring/src/main/java/com/bobocode/petros/bring/context/domain/BeanReference.java

```diff
@@ -0,0 +1,68 @@
+package com.bobocode.petros.bring.context.domain;
+
+import java.util.Objects;
+
+public class BeanReference {
+    private Object beanObject;
+    private boolean singleton = true;
+    private boolean prototype = false;
+
+    public Object getBeanObject() {
+        return beanObject;
+    }
+
+    public void setBeanObject(Object beanObject) {
+        this.beanObject = beanObject;
+    }
+
+    public boolean isSingleton() {
+        return singleton;
+    }
+
+    public void setSingleton(boolean singleton) {
+        this.singleton = singleton;
+    }
+
+    public boolean isPrototype() {
+        return prototype;
+    }
+
+    public void setPrototype(boolean prototype) {
+        this.prototype = prototype;
+    }
+
+    public BeanReference() {
```

Comment:
Class should have specific order, e.g. constants, fields, constructors, getters-setters, equals, hashcode, toString etc. May I ask you to put both constructors after the declared fields and before getters and setters?

Reply:
I would suggest adding lombok as a dependency to avoid all of that boilerplate code

Reply:
lombok is nice, I agree. But I thought we'll implement as much as we can without using it to practice implementing the things lombok provides out of the box.